.circleci/config.yml

```diff
@@ -1096,7 +1096,7 @@ jobs:
       ASAN_OPTIONS: check_initialization_order=true:detect_stack_use_after_return=true:strict_init_order=true:strict_string_checks=true:detect_invalid_pointer_pairs=2
       # Suppress CLN memory leak.
       # See: https://github.com/ethereum/solidity/issues/13891 for details.
-      LSAN_OPTIONS: suppressions=.circleci/cln-asan.supp
+      LSAN_OPTIONS: suppressions=.circleci/cln-asan.supp:print_suppressions=0
```

Comment:
So as you mentioned in your top level comment, this is probably not the best approach as it will just suppress the output, and not the actual issue (reflected by the non-zero error code). Is this built via `gcc` (I assume since `base_ubuntu2204`), and if yes, would it be possible to switch the `asan` job to `clang` since that would allow us to build with `-fsanitize-blacklist=<supressed>`, which should address the problem at its core (well, not at it its actual core, since it's still gonna leak, but at least we won't care).

Reply:
Wait, what's `.circleci/cln-asan.supp` and where can I find it? :D

Reply:
> So as you mentioned in your top level comment, this is probably not the best approach as it will just suppress the output, and not the actual issue (reflected by the non-zero error code). 

Yeah, but the output is what we evaluate in the cmdLineTests. The wrong exit code was just because in my previous attempt the suppression file was not being found.

> Is this built via `gcc` (I assume since `base_ubuntu2204`), and if yes, would it be possible to switch the `asan` job to `clang` since that would allow us to build with `-fsanitize-blacklist=<supressed>`, which should address the problem at its core (well, not at it its actual core, since it's still gonna leak, but at least we won't care).

Yes it is gcc. We already have asan for clang as well, but it doesn't run the cmdLineTests (https://github.com/ethereum/solidity/blob/develop/.circleci/config.yml#L1115) we could change it indeed. The error does not happen when using clang, so no need to suppress or blocklist it there.

Reply:
> Wait, what's `.circleci/cln-asan.supp` and where can I find it? :D

It is [here](https://github.com/ethereum/solidity/blob/develop/.circleci/cln-asan.supp). It is a file that you can inform to the AddressSanitizer to ignore certain errors. See here: https://github.com/google/sanitizers/wiki/AddressSanitizerLeakSanitizer#flags